go/api/response_types.go

```diff
@@ -88,33 +88,40 @@ const (
 // Enum-like structure which stores either a single-node response or multi-node response.
 // Multi-node response stored in a map, where keys are hostnames or "<ip>:<port>" strings.
 //
-// For example:
+// Example:
 //
 //	// Command failed:
 //	value, err := clusterClient.CustomCommand(args)
 //	value.IsEmpty(): true
 //	err != nil: true
 //
 //	// Command returns response from multiple nodes:
-//	value, _ := clusterClient.info()
-//	node, nodeResponse := range value.Value().(map[string]interface{}) {
-//	    response := nodeResponse.(string)
+//	value, _ := clusterClient.Info()
+//	for node, nodeResponse := range value.MultiValue() {
+//	    response := nodeResponse
 //	    // `node` stores cluster node IP/hostname, `response` stores the command output from that node
 //	}
 //
 //	// Command returns a response from single node:
-//	value, _ := clusterClient.infoWithRoute(Random{})
-//	response := value.Value().(string)
+//	value, _ := clusterClient.InfoWithOptions(api.ClusterInfoOptions{InfoOptions: nil, Route: api.RandomRoute.ToPtr()})
+//	response := value.SingleValue()
 //	// `response` stores the command output from a cluster node
 type ClusterValue[T any] struct {
 	valueType ValueType
-	value     Result[T]
+	value     Result[any]
```

Comment:
Can you clarify why this change is needed?

Reply:
In case when user runs `INFO` with multi-node route, response is a map (`map[string]string`), but we return `ClusterValue[string]`. So we have to store that map there.
https://github.com/valkey-io/valkey-glide/blob/eff425736388eda2880830f76c2ac34cc3f1155a/go/api/glide_cluster_client.go#L127

Maybe we should follow java approach and have 2 fields inside `ClusterValue`:
```go
type ClusterValue[T any] struct {
	valueType    ValueType
	singleValue  Result[T]
	multiValue   map[string]T
}
```
WDYT?

Reply:
Fixed